src/DependencyInjection/DrupalExtension.php

```diff
@@ -125,9 +125,11 @@ public function loadConfiguration(): void
             $name = "{$camelized}ServiceProvider";
             $class = "Drupal\\{$moduleName}\\{$name}";
 
-            if (class_exists($class)) {
-                $serviceClassProviders[$moduleName] = $class;
-            }
+            $serviceClassProviders[$moduleName] = $class;
+            $serviceId = "service_provider.$moduleName.service_provider";
+            $builder->parameters['drupalServiceMap'][$serviceId] = [
```

Comment:
so the variable`$serviceClassProviders` is never actually used. We could use that to bulk stuff in the `drupalServiceMap`.

Reply:
It is not, as mentioned in the beginning. Just left it here to avoid doing several things at once